homeassistant/components/matter/climate.py

```diff
@@ -422,6 +423,87 @@ def _get_temperature_in_degrees(
         return None
 
 
+class MatterTemperatureControlClimate(MatterEntity, ClimateEntity):
+    """Representation of a Matter climate entity."""
+
+    _attr_temperature_unit: str = UnitOfTemperature.CELSIUS
+    _attr_hvac_mode: HVACMode = HVACMode.HEAT_COOL
+    _attr_hvac_modes: list[HVACMode] = [HVACMode.HEAT_COOL]
+    _attr_target_temperature_step: float | None
+    _feature_map: int | None = None
+
+    _platform_translation_key = "thermostat"
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        target_hvac_mode: HVACMode | None = kwargs.get(ATTR_HVAC_MODE)
+        target_temperature: float | None = kwargs.get(ATTR_TEMPERATURE)
+
+        if target_hvac_mode is not None:
+            await self.async_set_hvac_mode(target_hvac_mode)
+
+        if target_temperature is not None:
+            # single setpoint control
+            if self.target_temperature != target_temperature:
+                await self.matter_client.send_device_command(
+                    node_id=self._endpoint.node.node_id,
+                    endpoint_id=self._endpoint.endpoint_id,
+                    command=clusters.TemperatureControl.Commands.SetTemperature(
+                        targetTemperature=int(
+                            target_temperature * TEMPERATURE_SCALING_FACTOR
+                        )
+                    ),
+                )
+            return
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update from device."""
+        self._calculate_features()
+
+        self._attr_hvac_action = None
+        # update target_temperature
+        self._attr_target_temperature = self._get_temperature_in_degrees(
+            clusters.TemperatureControl.Attributes.TemperatureSetpoint
+        )
+        # update min_temp
+        attribute = clusters.TemperatureControl.Attributes.MinTemperature
+        if (value := self._get_temperature_in_degrees(attribute)) is not None:
+            self._attr_min_temp = value
+        # update max_temp
+        attribute = clusters.TemperatureControl.Attributes.MaxTemperature
+        if (value := self._get_temperature_in_degrees(attribute)) is not None:
+            self._attr_max_temp = value
+
+    @callback
+    def _calculate_features(
+        self,
+    ) -> None:
+        """Calculate features for HA Thermostat platform from Matter FeatureMap."""
+        feature_map = int(
+            self.get_matter_attribute_value(
+                clusters.TemperatureControl.Attributes.FeatureMap
+            )
+        )
+        self._feature_map = feature_map
```

Comment:
Is it needed to calculate the features every time ? Maybe just store the previous value of the featuremap into a variable so you can compare that to determine if the features changed, see some of the other platforms for examples